compiler/passes/normalize.cpp

```diff
@@ -4215,7 +4253,10 @@ static void fixupQueryFormals(FnSymbol* fn) {
 
         replaceUsesWithPrimTypeof(fn, formal);
 
-      } else if (isQueryForGenericTypeSpecifier(formal) == true) {
+      } else if (isCastToBorrowedInFormal(formal)) {
```

Comment:
is a similar thing necessary for `shared` `owned` or `unmanaged`? Or is the intention to leave those for future work? If it's future work, add a TODO comment here & mention it in the PR message, please